app/scripts/background.js

```diff
@@ -1144,8 +1138,22 @@ export function setupController(
     preinstalledSnaps,
     requestSafeReload,
     cronjobControllerStorageManager,
+    // set to false because we need to subscribe to some events *before*
+    // configuring controllers to ensure we are able to record state changes
+    autoConfigureControllers: false,
   });
 
+  // this persistence hooks must happen before `controller.configureAllControllers`
+  // or some update might not be persisted!
+  controller.store.on('update', update);
```

Comment:
I think I understand why those Sentry tests are failing now. This is unsafe to do right now because we'll end up persisting partway through initialization, when not all controllers are constructed yet. This is OK after we do state splitting, but until then we're capturing a "complete" state snapshot each time (or at least attempting to), but it's not really complete because we're mid-init.

If we want to introduce this before state splitting, I think we'll need a guard against this running before controller initialization has finished.

Reply:
AH, this is just for the _sentry_ part. Persistence is fine, right?

Is this only "unsafe" for the tests?